modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java

```diff
@@ -0,0 +1,593 @@
+        Optional<DistributionSnapshot> distribution = snaps.stream()
+                                                           .filter(snap -> snap.getName()
+                                                                               .toLowerCase()
+                                                                               .startsWith("nuxeo platform"))
+                                                           .findFirst();
+
+        String latest = "current";
+        if (distribution.isPresent()) {
+            latest = distribution.get().getKey();
+        }
+        return ctx.newObject("redirectWO", "latest", latest);
+    }
+
+    @Path("{distributionId}")
+    public Resource viewDistribution(@PathParam("distributionId") String distributionId) {
+        if (distributionId == null || "".equals(distributionId)) {
+            return this;
+        }
+
+        List<DistributionSnapshot> snaps = getSnapshotManager().listPersistentSnapshots((ctx.getCoreSession()));
+        if (distributionId.matches(VERSION_REGEX.toString())) {
+            String finalDistributionId = distributionId;
+            String distribution = snaps.stream()
+                                       .filter(s -> s.getVersion().equals(finalDistributionId))
+                                       .findFirst()
+                                       .map(DistributionSnapshot::getKey)
+                                       .orElse("current");
+
+            return ctx.newObject("redirectWO", finalDistributionId, distribution);
+        }
+
+        String orgDistributionId = distributionId;
+        Boolean embeddedMode = Boolean.FALSE;
+        if ("adm".equals(distributionId)) {
+            embeddedMode = Boolean.TRUE;
+        } else {
+            snaps.add(getSnapshotManager().getRuntimeSnapshot());
+            distributionId = SnapshotResolverHelper.findBestMatch(snaps, distributionId);
+        }
+        if (distributionId == null || "".equals(distributionId)) {
+            distributionId = "current";
+        }
+
+        if (!orgDistributionId.equals(distributionId)) {
+            return ctx.newObject("redirectWO", orgDistributionId, distributionId);
+        }
+
+        ctx.setProperty("embeddedMode", embeddedMode);
+        ctx.setProperty("distribution", getSnapshotManager().getSnapshot(distributionId, ctx.getCoreSession()));
+        ctx.setProperty(DIST_ID, distributionId);
+        return ctx.newObject("apibrowser", distributionId, embeddedMode);
+    }
+
+    public List<DistributionSnapshotDesc> getAvailableDistributions() {
+        return getSnapshotManager().getAvailableDistributions(ctx.getCoreSession());
+    }
+
+    public String getRuntimeDistributionName() {
+        return SnapshotManagerComponent.RUNTIME;
+    }
+
+    public DistributionSnapshot getRuntimeDistribution() {
+        return getSnapshotManager().getRuntimeSnapshot();
+    }
+
+    public List<DistributionSnapshot> listPersistedDistributions() {
```

Comment:
The only usage I see is in `getLatest` which turns back the result to stream, should this method return `Stream`?

Reply:
indeed but i only moved that code -> i would prefer not modifying it to avoid unforeseen regressions in these logics